### DISCUSSION TITLE
"Move Minimize/Restore/Close window buttons to the right of Hamburger menu" solution is broken since Firefox 99

### DISCUSSION THREAD

**Discussion by haydenseitz** [piroor/treestyletab#3110] [General]
## Short description
When using userChrome customizations for [Hide horizontal tabs at the top of the window](https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#hide-horizontal-tabs-at-the-top-of-the-window-1349-1672-2147) and [Move Minimize/Restore/Close window buttons to the right of Hamburger menu ](https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#move-minimizerestoreclose-window-buttons-to-the-right-of-hamburger-menu-after-hiding-horizontal-tabs), the top of the titlebar can no longer be grabbed to drag the window as of Firefox v99. I know this is not an issue with TST, but I'm hoping someone from the community can find a solution where I cannot 😊. My current workaround is to remove this section of code:

```css
/* Fix for main menu calling by Alt button */
#titlebar > #toolbar-menubar {
	margin-top: 10px;
}
```

However, the main menu is cut off at the top when pressing Alt:

![image](https://user-images.githubusercontent.com/17074342/164284510-b226df97-6111-4864-abb3-3e9d98168d67.png)

## Steps to reproduce

 1. Start Firefox **_v99_** with clean profile.
 2. Install TST.
 3. Add userChrome styling for [Hide horizontal tabs at the top of the window](https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#hide-horizontal-tabs-at-the-top-of-the-window-1349-1672-2147) and [Move Minimize/Restore/Close window buttons to the right of Hamburger menu ](https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#move-minimizerestoreclose-window-buttons-to-the-right-of-hamburger-menu-after-hiding-horizontal-tabs)
 4. Try to drag window by selecting top of titlebar

## Expected result
Able to select and drag the top of the titlebar when using the customizations

## Actual result
Cannot grab the top of the titlebar (but can grab it ~10px down from the top)

## Environment

 * Platform (OS): Windows
 * Version of Firefox:  99.0.1
 * Version (or revision) of Tree Style Tab:  3.8.21

**Comment by piroor** [comment]
(I've moved this from issues to discussions, because it is out of scope of TST project itself. It is a common knowledge about customizing Firefox with userChrome.css.)

**Comment by irvinm** [comment]
Try giving this a shot ... not sure if it is exactly what you wanted.

```
#main-window[tabsintitlebar="true"]:not([extradragspace="true"]) #TabsToolbar > .toolbar-items {
	opacity: 0;
	pointer-events: none;
}

#nav-bar {
	margin-right:140px;
}

#titlebar {
	appearance: none !important;
	height: 0px; 
}

#TabsToolbar > .titlebar-buttonbox-container {
	display: block;
	position: absolute;
	top: 7px;
	right: 1px;
}
```

![image](https://user-images.githubusercontent.com/979729/164340275-0ed1480d-2e34-4a4d-99a3-f1e4cabd3797.png)

![image](https://user-images.githubusercontent.com/979729/164340282-ba6e63d8-d31e-4422-afaf-9e0a6e52da67.png)

**Reply by haydenseitz** [reply]
that's close, but for me the top of the titlebar is cutoff when viewing maximized window. I'll try to tweak to find a solution

**Reply by irvinm** [reply]
Ah, I see what you are saying ... I see the same issue when maximized.  I bet there is a way to add some custom CSS "spacing" only when the window is maximized.

**Reply by irvinm** [reply]
Adding this worked for me when maximized.

`:root[tabsintitlebar][sizemode="maximized"] { padding-top: 8px !important; }`
